src/hooks/handleQueryInUrl.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  usePageContext,
+  usePanelContext,
+  useQueriesCurrentForPage,
+} from "@/hooks";
+import type { OracleQueryUI } from "@/types";
+import filter from "lodash/filter";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+
+export function useHandleQueryInUrl() {
```

Comment:
logic is the same, just renamed to prevent confusion with other page-related hooks